akka-docs-dev/rst/scala/code/docs/stream/cookbook/RecipeReduceByKey.scala

```diff
@@ -17,22 +17,15 @@ class RecipeReduceByKey extends RecipeSpec {
       def words = Source(List("hello", "world", "and", "hello", "universe", "akka") ++ List.fill(1000)("rocks!"))
 
       //#word-count
-      // split the words into separate streams first
-      val wordStreams: Source[(String, Source[String, Unit]), Unit] = words.groupBy(identity)
-
-      // add counting logic to the streams
-      val countedWords: Source[Future[(String, Int)], Unit] = wordStreams.map {
-        case (word, wordStream) =>
-          wordStream.runFold((word, 0)) {
-            case ((w, count), _) => (w, count + 1)
-          }
-      }
-
-      // get a stream of word counts
-      val counts: Source[(String, Int), Unit] =
-        countedWords
-          .buffer(MaximumDistinctWords, OverflowStrategy.fail)
-          .mapAsync(4)(identity)
+      val counts: Source[(String, Int), Unit] = words
+        // split the words into separate streams first
+        .groupBy(identity)
+        // add counting logic to the streams
+        .fold(("", 0)) {
```

Comment:
another reason why a non-terminal `fold` is useful. Thanks @viktorklang for doing the necessary refactor!